funnel/models/commentvote.py

```diff
@@ -113,6 +113,30 @@ def __init__(self, **kwargs):
         super(Commentset, self).__init__(**kwargs)
         self.count = 0
 
+    @property
+    def parent(self):
+        parent = None  # project or proposal object
+        if hasattr(self, 'project') and self.project:
+            parent = self.project
+        elif hasattr(self, 'proposal') and self.proposal:
+            parent = self.proposal
+        return parent
+
+    @property
+    def parent_commentset_url(self):
```

Comment:
This is a bit cringe-y, but I don't have a better idea. One possibility is to make it a view function, so it will be `commentset.views.url()`